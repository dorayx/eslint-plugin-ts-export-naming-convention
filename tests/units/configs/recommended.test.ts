import { resolve, join } from 'node:path';
import { fileURLToPath } from 'node:url';
import { chdir, cwd } from 'node:process';
import { afterEach, beforeEach, describe, expect } from 'vitest';
import { ESLint } from 'eslint';

const curFilePath = fileURLToPath(import.meta.url);
const originalCwd = cwd();
const fixturesRoot = resolve(curFilePath, '../../../fixtures/configs');
const pkgRoot = resolve(curFilePath, '../../../../dist');

describe('configs/recommended', () => {
  describe('in an ESM package', (it) => {
    let linter: ESLint;

    beforeEach(() => {
      chdir(join(fixturesRoot, 'esm'));
      linter = new ESLint({
        baseConfig: {
          extends: ['plugin:@typescript-eslint/recommended', join(pkgRoot, 'configs/recommended.js')],
          parser: '@typescript-eslint/parser',
          rules: {
            // for testing only
            '@typescript-eslint/no-unused-vars': 'off',
          },
        },
        useEslintrc: false,
      });
    });

    afterEach(() => {
      chdir(originalCwd);
    });

    it('validates pascal case', async () => {
      const report = await linter.lintText(
        `
      export interface ITestSomething {}      
      export interface iTestSomething {}
      export interface I_TestSomething {}
      
      export type TTestSomething = { prop: 'value' };
      export type tTestSomething = { prop: 'value' };
      export type T_TestSomething = { prop: 'value' };
      `,
        {
          filePath: join(fixturesRoot, 'test.js'),
        },
      );

      expect(report[0].messages).toMatchSnapshot();
    });

    it('validates interface prefixes', async () => {
      const report = await linter.lintText(
        `
      export interface ITestSomething {}
      interface TestSomething {}
      `,
        {
          filePath: join(fixturesRoot, 'test.js'),
        },
      );

      expect(report[0].messages).toMatchSnapshot();
    });

    it('validates type aliase prefixes', async () => {
      const report = await linter.lintText(
        `
      export type TTestSomething = { prop: 'value' };
      type TestSomething = { prop: 'value' };
      `,
        {
          filePath: join(fixturesRoot, 'test.js'),
        },
      );

      expect(report[0].messages).toMatchSnapshot();
    });
  });
});
