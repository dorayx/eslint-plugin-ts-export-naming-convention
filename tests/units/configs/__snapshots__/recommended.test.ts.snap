// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`configs/recommended > in an ESM package > validates interface prefixes 1`] = `[]`;

exports[`configs/recommended > in an ESM package > validates pascal case 1`] = `
[
  {
    "column": 24,
    "endColumn": 38,
    "endLine": 3,
    "line": 3,
    "message": "Interface name \`iTestSomething\` must have one of the following prefixes: I",
    "messageId": "missingAffix",
    "nodeType": "Identifier",
    "ruleId": "@typescript-eslint/naming-convention",
    "severity": 2,
  },
  {
    "column": 24,
    "endColumn": 39,
    "endLine": 4,
    "line": 4,
    "message": "Interface name \`I_TestSomething\` trimmed as \`_TestSomething\` must match one of the following formats: PascalCase",
    "messageId": "doesNotMatchFormatTrimmed",
    "nodeType": "Identifier",
    "ruleId": "@typescript-eslint/naming-convention",
    "severity": 2,
  },
  {
    "column": 19,
    "endColumn": 33,
    "endLine": 7,
    "line": 7,
    "message": "Type Alias name \`tTestSomething\` must have one of the following prefixes: T",
    "messageId": "missingAffix",
    "nodeType": "Identifier",
    "ruleId": "@typescript-eslint/naming-convention",
    "severity": 2,
  },
  {
    "column": 19,
    "endColumn": 34,
    "endLine": 8,
    "line": 8,
    "message": "Type Alias name \`T_TestSomething\` trimmed as \`_TestSomething\` must match one of the following formats: PascalCase",
    "messageId": "doesNotMatchFormatTrimmed",
    "nodeType": "Identifier",
    "ruleId": "@typescript-eslint/naming-convention",
    "severity": 2,
  },
]
`;

exports[`configs/recommended > in an ESM package > validates type aliase prefixes 1`] = `[]`;
